contracts/dex_proxy_contract.py

```diff
@@ -455,6 +454,15 @@ def add_farm_to_intermediate(self, deployer: Account, proxy: ProxyNetworkProvide
 
         gas_limit = 50000000
         return endpoint_call(proxy, gas_limit, deployer, Address(self.address), "addFarmToIntermediate", [farm_address])
+    
+    def add_contract_to_whitelist(self, deployer: Account, proxy: ProxyNetworkProvider, whitelisted_sc_address: str):
```

Comment:
I came up with BaseSCWhitelistContract a few days later out of annoyance and I haven't propagated it entirely. 
I have to put it in proxy dex and energy factory as well, then refactor the existing whitelist usages in staking/metastaking to use this one, since they're basically the same.